cmd/bucket-replication.go

```diff
@@ -2475,3 +2483,195 @@ func queueReplicationHeal(ctx context.Context, bucket string, oi ObjectInfo, rcf
 	}
 	return
 }
+
+const mrfTimeInterval = 5 * time.Minute
+
+func (p *ReplicationPool) persistMRF() {
+	var mu sync.Mutex
+	entries := make(map[string]MRFReplicateEntry)
+	mTimer := time.NewTimer(mrfTimeInterval)
+	defer mTimer.Stop()
+	saveMRFToDisk := func(drain bool) {
+		mu.Lock()
+		defer mu.Unlock()
+		if len(entries) == 0 {
+			return
+		}
+		cctx := p.ctx
+		if drain {
+			cctx = context.Background()
+			// drain all mrf entries and save to disk
+			for e := range p.mrfSaveCh {
+				entries[e.versionID] = e
+			}
+		}
+		if err := p.saveMRFEntries(cctx, entries); err != nil {
+			logger.LogOnceIf(p.ctx, fmt.Errorf("Unable to persist replication failures to disk:%w", err), string(replicationSubsystem))
+		}
+		entries = make(map[string]MRFReplicateEntry)
+		return
+	}
+	for {
+		select {
+		case <-mTimer.C:
+			saveMRFToDisk(false)
+			mTimer.Reset(mrfTimeInterval)
+		case <-p.ctx.Done():
+			p.mrfOnce.Do(
+				func() {
+					close(p.mrfSaveCh)
+				})
+			saveMRFToDisk(true)
+			return
+		case e, ok := <-p.mrfSaveCh:
+			if !ok {
+				return
+			}
+			var cnt int
+			mu.Lock()
+			entries[e.versionID] = e
+			cnt = len(entries)
+			mu.Unlock()
+			if cnt >= cap(p.mrfSaveCh) || len(p.mrfSaveCh) >= cap(p.mrfSaveCh) {
```

Comment:
If this condition is true, it means that MRF is already overwhelmed and most likley p.mrfSaveCh will continue to be full all the time, that also means we will try to save in the disk for each element from p.mrfSaveCh. I guess let's try to save in the disk before the channel gets full (cap(p.mrfSaveCh)*0.8)

Reply:
good suggestion - done.